### PR TITLE
Implement Brotli support

### PR DESCRIPTION
**Pull request by alxpy**
## What do these changes do?
 Implement [Brotli](https://github.com/google/brotli) support

## Are there changes in behavior for the user?
No.

## Related issue number
https://github.com/aio-libs/aiohttp/issues/2270

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
- [x] Add a new news fragment into the `changes` folder

### CODE REVIEW

**Comment by asvetlov** [comment]
Good start but tests are required.

**Comment by oleksandr-kuzmenko** [comment]
I hope this is not a big problem
`Coverage   97.23%   97.18%   -0.05% `

**Comment by asvetlov** [comment]
I pretty happy with the value.
Please add `# pragma: no cover` for lines which are executed if no `brotlipy` installed.

**Comment by hellysmile** [comment]
Hello there, recently I was playing with exactly brotli and were using official python package.

Is there any difference between https://pypi.python.org/pypi/Brotli and https://pypi.python.org/pypi/brotlipy

Can we support both of them?

**Comment by oleksandr-kuzmenko** [comment]
@hellysmile hello
I can't find docs for google `brotli` py-package =(

`Brotlipy` is a collection of bindings to the Brotli C library by Google, has docs and:
> Brotlipy has a very similar interface to the standard library’s zlib module

I think no:
> Can we support both of them?

**Comment by oleksandr-kuzmenko** [comment]
`Coverage   97.23%   97.23%   +<.01%`
FYI @asvetlov

**Comment by oleksandr-kuzmenko** [comment]
In the near future I will try to add brotli support to server side

**Comment by asvetlov** [comment]
I mean now we do support brotli decompression only, compressing requests/responses would be very useful too.

**Comment by oleksandr-kuzmenko** [comment]
I got what you meant

**Comment by kxepal** [comment]
awesome!

**Comment by hellysmile** [comment]
I've tried to play around brotli\brotlipy, at least test is green on both of them.

**Comment by asvetlov** [comment]
Looks like we need mention both libraries in doc

**Comment by oleksandr-kuzmenko** [comment]
ok, i'll update docs

File: aiohttp/http_parser.py [hunk 1]
```diff
@@ -605,23 +612,29 @@ def __init__(self, out, encoding):
         self.encoding = encoding
         self._started_decoding = False
 
-        zlib_mode = (16 + zlib.MAX_WBITS
-                     if encoding == 'gzip' else -zlib.MAX_WBITS)
-
-        self.zlib = zlib.decompressobj(wbits=zlib_mode)
+        if encoding == 'br':
+            if not HAS_BROTLI:  # pragma: no cover
```
Comment (webknjaz) [inline aiohttp/http_parser.py:616]: Why this has `no cover`? Good coverage matters

**Comment by asvetlov** [comment]
We don't use extras in aiohttp (but I like the idea).
Hmm, we should mention `brotly`/`brotlipy` as optional dependency in `index.rst` and `README.rst` along with `cchardet` and `aiodns`.

**Pull request merged**